library/src/main/java/com/chuckerteam/chucker/internal/support/Sharable.kt

```diff
@@ -0,0 +1,72 @@
+package com.chuckerteam.chucker.internal.support
+
+import android.app.Activity
+import android.content.ClipData
+import android.content.Context
+import android.content.Intent
+import android.widget.Toast
+import androidx.core.app.ShareCompat
+import androidx.core.content.FileProvider
+import com.chuckerteam.chucker.R
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.withContext
+
+internal interface Sharable {
+    fun toSharableContent(context: Context): String
```

Comment:
I'm not very fond of passing `Context` here. It makes implementations that do not use Android (like cURL one) dependent on it and a little bit harder to test.

On the other hand passing `Context` as an argument to the implementation might lead to leaks (even though it wouldn't do it ATM).

Reply:
I don't think that passing `Context` here is an issue now. We can always reiterate on this decision later.